cmd/jujud/agent/machine.go

```diff
@@ -900,11 +899,6 @@ func (a *MachineAgent) setupContainerSupport(st api.Connection, logger machine.L
 	m := result[0].Machine
 
 	var supportedContainers []instance.ContainerType
-	supportsContainers := container.ContainersSupported()
-	if supportsContainers {
-		supportedContainers = append(supportedContainers, instance.LXD)
```

Comment:
You need to keep this line unconditionally.

Reply:
In 4.0 you should blow this method away and just get rid of the container support methods completely.